packages/blob/src/index.js

```diff
@@ -70,3 +70,41 @@ export function isBlobURL( url ) {
 	}
 	return url.indexOf( 'blob:' ) === 0;
 }
+
+/**
+ * Downloads a file, e.g., a text or readable stream, in the browser.
+ * Appropriate for downloading smaller file sizes, e.g., < 5 MB.
+ *
+ * Example usage:
+ *
+ * ```js
+ * 	const fileContent = JSON.stringify(
+ * 		{
+ * 			"title": "My Post",
+ * 		},
+ * 		null,
+ * 		2
+ * 	);
+ * 	const fileName = 'file.json';
+ *
+ * 	downloadBlob( 'file.json', fileContent, 'application/json' );
+ * ```
+ *
+ * @param {string}   fileName    File Name.
+ * @param {BlobPart} content     File Content (BufferSource | Blob | string).
+ * @param {string}   contentType File mime type.
+ * @return {HTMLElement}        The created anchor element.
+ */
+export function downloadBlob( fileName = 'wp-download', content, contentType ) {
+	const file = new window.Blob( [ content ], { type: contentType } );
+	const url = window.URL.createObjectURL( file );
+	const a = document.createElement( 'a' );
+	a.href = url;
+	a.download = fileName;
+	a.style.display = 'none';
+	document.body.appendChild( a );
+	a.click();
+	document.body.removeChild( a );
+	window.URL.revokeObjectURL( url );
+	return a;
```

Comment:
Any reason why we want to return the anchor element instead of removing it to avoid memory leak?

Reply:
Thanks for reviewing! 

By "removing" do you mean passively via garbage collection, or something other than removing from DOM?

I returned for two reasons:

1. To have something the tests can easily verify
2. Just in case consumers what to do something with a link element

Reason 2 is not a very good reason though 😄

Reply:
I feel like the anchor element is an implementation detail and shouldn't be exposed to the consumers. What the consumers expect is just a function which triggers downloading something. The same goes with testing, we should avoid testing implementation details if possible (probably a bit trickier here 😅). If the user keeps the returned element then it could be a source of memory leak (though doesn't seem that impactful).

Reply:
I agree. I'll remove. Thanks!

Reply:
> The same goes with testing, we should avoid testing implementation details if possible (probably a bit trickier here)

You were right, but I think managed to hack together a replacement test 🙃

Reply:
Yeah I guess it doesn't really matter here if we have a solid unit test. Whatever method we end up with will have to rely on mocking implementation details in jsdom. A potential better method would be doing it in integration testing with a real browser for instance, but that's out of the scope and probably not worth it here.